src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs

```diff
@@ -71,5 +72,32 @@ public static void AssertRoundtrips<T>(T expected, params Func<T, object>[] addi
                 }
             }
         }
+
+        public static void AssertExceptionDeserializationFails<T>() where T : Exception
+        {
+            // .NET Core throws a PlatformNotSupportedException for deserializing many exceptions.
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
+            {
+                return;
+            }
+
+            var info = new SerializationInfo(typeof(T), new FormatterConverter());
+            info.AddValue("ClassName", "ClassName");
+            info.AddValue("Message", "Message");
+            info.AddValue("InnerException", null);
+            info.AddValue("HelpURL", null);
+            info.AddValue("StackTraceString", null);
+            info.AddValue("RemoteStackTraceString", null);
+            info.AddValue("RemoteStackIndex", 5);
+            info.AddValue("HResult", 5);
+            info.AddValue("Source", null);
+            info.AddValue("ExceptionMethod", null);
+
+            ConstructorInfo constructor = typeof(T).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, new Type[] { typeof(SerializationInfo), typeof(StreamingContext) }, null);
```

Comment:
@AtsushiKan , correct, that update isn't done yet. Additionally, if the serialization constructors are private, they will be reflection blocked, so you won't be able to find them.